index.js

```diff
@@ -76,13 +82,76 @@ if (process.platform === 'win32') {
 
 getFn = function () {
 	var fn = function (/*…msg*/) {
-		var start = '', end = '';
-		forEach(fn._cliColorData, function (mod) {
-			end = '\x1b[' + mod[1] + 'm' + end;
-			start += '\x1b[' + mod[0] + 'm';
-		}, null, true);
-		return start + join.call(arguments, ' ') + end;
+		var start = ''
+		  , end   = ''
+		  , keys  = Object.keys(fn._cliColorData)
+		  , msg   = join.call(arguments, ' ');
+
+		if (keys.length) {
+			for (var i=0; i<keys.length; i++) {
+				start += ( i !== 0 ? ';' : '' ) + fn._cliColorData[keys[i]][0];
+				end   += ( i !== 0 ? ';' : '' ) + fn._cliColorData[keys[i]][1];
```

Comment:
In this case is not needly a micro-optimization. It's for avoid code redundancy. But probably I'll change it because of next pharagraph.

Well, about join commands by `;`, as I tried only on Windows, it works fine, but really it need be checked nearly. Possiblely, on Unix it doesn't works -- based on @StreetStrider tests above. So in this case, really the best way is return to old method. And will be better because of new method of parse nested color. Will be more accurate, in case.